File: Sample Projects/VWMasterApplication/VWMasterApplication/VWAppNexusServerViewController.m
//
//  VWAppNexusServerViewController.m
//  VWMasterApplication
//
//  Created by Danijel Lombarovic on 20/09/16.
//  Copyright © 2016 Christina Long. All rights reserved.
//

#import "VWAppNexusServerViewController.h"
#import "ANBannerAdView.h"
#import "ANInterstitialAd.h"
#import <VerveAd/VerveAd.h>
#import "ANSDKSettings.h"

static NSString *const kPlacementIdBanner = @"11113606";
static NSString *const kPlacementIdBannerTablet = @"11113608";
static NSString *const kPlacementIdInline = @"11113607";
static NSString *const kPlacementIdInterstitial = @"11113611";

@interface VWAppNexusServerViewController () <ANBannerAdViewDelegate, ANInterstitialAdDelegate>

@property (strong, nonatomic) ANBannerAdView *bannerAdView;
@property (strong, nonatomic) ANBannerAdView *inlineAdView;
@property (strong, nonatomic) ANInterstitialAd *interstitialAdView;

@end

@implementation VWAppNexusServerViewController

#pragma mark - View Controller Lifecycle

- (void)viewDidLoad {
  [super viewDidLoad];
  
  BOOL canUseHTTP = NO;
  NSDictionary *appTransportSecurity =  [[[NSBundle mainBundle] infoDictionary] valueForKey:@"NSAppTransportSecurity"];
  
  if ([[UIDevice currentDevice].systemVersion floatValue] >= 10.0) {
    canUseHTTP = !([appTransportSecurity valueForKey:@"NSAllowsArbitraryLoadsInWebContent"] || [appTransportSecurity valueForKey:@"NSAllowsArbitraryLoadsInMedia"] || [appTransportSecurity valueForKey:@"NSAllowsLocalNetworking"]) && [[appTransportSecurity valueForKey:@"NSAllowsArbitraryLoads"] boolValue];
  } else {
    canUseHTTP = [[appTransportSecurity valueForKey:@"NSAllowsArbitraryLoads"] boolValue];
  }
  
  [ANSDKSettings sharedInstance].HTTPSEnabled = !canUseHTTP;

  
  [self addBannerAdView];
  [self addInlineAdView];
}


#pragma mark - Banner Ad

- (void)addBannerAdView {
 
  //determine size and farame for banner ad
  CGRect bounds = self.view.bounds;
  CGSize adSize = UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad ? CGSizeMake(728, 90) : CGSizeMake(320, 50);
  
  CGRect adFrame = CGRectMake((bounds.size.width - adSize.width)/2, bounds.size.height - adSize.height - self.tabBarController.tabBar.frame.size.height, adSize.width, adSize.height);
  
  NSString *placementID = UI_USER_INTERFACE_IDIOM() == UIUserInterfaceIdiomPad ? kPlacementIdBannerTablet : kPlacementIdBanner;
  self.bannerAdView = [ANBannerAdView adViewWithFrame:adFrame placementId:placementID adSize:adSize];
  self.bannerAdView.rootViewController = self;
  self.bannerAdView.autoRefreshInterval = 0;
  self.bannerAdView.delegate = self;
  self.bannerAdView.backgroundColor = [UIColor grayColor];
  
  //ad the banner ad to the view
  [self.view addSubview:self.bannerAdView];
}

- (IBAction)requestBannerAd:(id)sender {
  [self.bannerAdView loadAd];
}


#pragma mark - Inline Ad

- (void)addInlineAdView {

  //determine size and frame for the inline ad
  CGRect bounds = self.view.bounds;
  CGSize adSize = CGSizeMake(300, 250);
  
  CGRect adFrame = CGRectMake((bounds.size.width - adSize.width)/2, (bounds.size.height - adSize.height)/2, adSize.width, adSize.height);
  
  self.inlineAdView = [ANBannerAdView adViewWithFrame:adFrame placementId:kPlacementIdInline adSize:adSize];
  self.inlineAdView.rootViewController = self;
  self.inlineAdView.autoRefreshInterval = 0;
  self.inlineAdView.delegate = self;
  self.inlineAdView.backgroundColor = [UIColor grayColor];
  
  //add the inline ad to the view
  [self.view addSubview:self.inlineAdView];
}

- (IBAction)requestInlineAd:(id)sender {
  [self.inlineAdView loadAd];
}


#pragma mark - Interstitial Ad

- (IBAction)requestInterstitialAd:(id)sender {
  self.interstitialAdView = [[ANInterstitialAd alloc] initWithPlacementId:kPlacementIdInterstitial];
  self.interstitialAdView.delegate = self;
  [self.interstitialAdView loadAd];
}


#pragma mark - ANBannerAdViewDelegate

- (void)adDidReceiveAd:(id<ANAdProtocol>)ad
{
  NSLog(@"adDidReceiveAd: %@", [ad placementId]);
  if ([ad isEqual:self.interstitialAdView]) {
    [self.interstitialAdView displayAdFromViewController:self];
  }
}

- (void)ad:(id<ANAdProtocol>)ad requestFailedWithError:(NSError *)error
{
  NSLog(@"ad:requestFailedWithError: %@", [ad placementId]);
}

- (void)adWasClicked:(id<ANAdProtocol>)ad
{
  NSLog(@"adWasClicked: %@", [ad placementId]);
}

- (void)adWillClose:(id<ANAdProtocol>)ad
{
  NSLog(@"adWillClose: %@", [ad placementId]);
}

- (void)adDidClose:(id<ANAdProtocol>)ad
{
  NSLog(@"adDidClose: %@", [ad placementId]);
}

- (void)adWillPresent:(id<ANAdProtocol>)ad
{
  NSLog(@"adWillPresent: %@", [ad placementId]);
}

- (void)adDidPresent:(id<ANAdProtocol>)ad
{
  NSLog(@"adDidPresent: %@", [ad placementId]);
}

- (void)adWillLeaveApplication:(id<ANAdProtocol>)ad
{
  NSLog(@"adWillLeaveApplication: %@", [ad placementId]);
}


#pragma mark - ANInterstitialAdDelegate

- (void)adFailedToDisplay:(ANInterstitialAd *)ad
{
  NSLog(@"adFailedToDisplay: %@", [ad placementId]);
}

@end

File: Sample Projects/VWAdSwiftExample/VWAdSwiftExample/VWAppNexusServerViewController.swift
//
//  VWAppNexusServerViewController.swift
//  VWAdSwiftExample
//
//  Created by Danijel Lombarovic on 20/09/16.
//  Copyright © 2016 Verve Wireless. All rights reserved.
//

import UIKit

class VWAppNexusServerViewController: UIViewController, ANBannerAdViewDelegate, ANInterstitialAdDelegate {
  
  let kPlacementIdBanner = "11113606"
  let kPlacementIdBannerTablet = "11113608"
  let kPlacementIdInline = "11113607"
  let kPlacementIdInterstitial = "11113611";
  
  var bannerAdView: ANBannerAdView?
  var inlineAdView: ANBannerAdView?
  var interstitialAdView: ANInterstitialAd?
		
  //MARK: View Controller Lifecycle
  
  override func viewDidLoad() {
    super.viewDidLoad()
    
    var canUseHTTP: Bool = false
    guard let infoDictionary = Bundle.main.infoDictionary else { return }
    if let appTransportSecurity = infoDictionary["NSAppTransportSecurity"] as? NSDictionary {
      
      let allowsArbitraryLoadsInWebContent = appTransportSecurity.object(forKey: "NSAllowsArbitraryLoadsInWebContent") != nil
      let allowsArbitraryLoadsInMedia = appTransportSecurity.object(forKey: "NSAllowsArbitraryLoadsInMedia") != nil
      let allowsLocalNetworking = appTransportSecurity.object(forKey: "NSAllowsLocalNetworking") != nil
      
      var allowsArbitraryLoadsValue: Bool = false
      if let allowsArbitraryLoads = appTransportSecurity.object(forKey: "NSAllowsArbitraryLoads") as? Bool {
        allowsArbitraryLoadsValue = allowsArbitraryLoads
      }

      if UIDevice.current.systemVersion.compare("10.0", options: .numeric) == .orderedDescending {
        canUseHTTP = !(allowsArbitraryLoadsInWebContent || allowsArbitraryLoadsInMedia || allowsLocalNetworking) && allowsArbitraryLoadsValue
      } else {
        canUseHTTP = allowsArbitraryLoadsValue
      }
    }
    
    ANSDKSettings.sharedInstance().httpsEnabled = !canUseHTTP
    
    addBannerAdView()
    addInlineAdView()
  }
  
  
  //MARK: Banner Ad
  
  func addBannerAdView() {
    //determine size and farame for banner ad
    let bounds = view.bounds
    let adSize = UI_USER_INTERFACE_IDIOM() == .pad ? CGSize(width: 728, height: 90) : CGSize(width: 320, height: 50)
    
    guard let tabBarHeight = tabBarController?.tabBar.frame.size.height else {
      return
    }
    let adFrame = CGRect(
      x: (bounds.size.width - adSize.width)/2,
      y: bounds.size.height - adSize.height - tabBarHeight,
      width: adSize.width,
      height: adSize.height)
    
    let placementId = UI_USER_INTERFACE_IDIOM() == .pad ? kPlacementIdBannerTablet : kPlacementIdBanner
    guard let bannerAdView = ANBannerAdView(frame: adFrame, placementId: placementId, adSize: adSize) else { return }
    bannerAdView.rootViewController = self
    bannerAdView.autoRefreshInterval = 0
    bannerAdView.delegate = self
    bannerAdView.backgroundColor = .gray
    
    //ad the banner ad to the view
    view.addSubview(bannerAdView)
    
    self.bannerAdView = bannerAdView
  }
  
  @IBAction func requestBannerAd() {
    bannerAdView?.loadAd()
  }
  
  //MARK: Inline Ad
  
  func addInlineAdView() {
    //determine size and frame for the inline ad
    let bounds = view.bounds
    let adSize = CGSize(width: 300, height: 250)
    
    let adFrame = CGRect(
      x: (bounds.size.width - adSize.width)/2,
      y: (bounds.size.height - adSize.height)/2,
      width: adSize.width,
      height: adSize.height)
    
    guard let inlineAdView = ANBannerAdView(frame: adFrame, placementId: kPlacementIdInline, adSize: adSize) else { return }
    inlineAdView.rootViewController = self
    inlineAdView.autoRefreshInterval = 0
    inlineAdView.delegate = self
    inlineAdView.backgroundColor = .gray
    
    //add the inline ad to the view
    view.addSubview(inlineAdView)
    
    self.inlineAdView = inlineAdView
  }
  
  @IBAction func requestInlineAd() {
    inlineAdView?.loadAd()
  }
  
  //MARK: Interstitial Ad
  
  @IBAction func requestInterstitialAd() {
    interstitialAdView = ANInterstitialAd(placementId: kPlacementIdInterstitial)
    interstitialAdView?.delegate = self
    interstitialAdView?.load()
  }
  
  //MARK: ANBannerAdViewDelegate
  
  func adDidReceiveAd(_ ad: ANAdProtocol!) {
    NSLog("adDidReceiveAd: \(ad.placementId)")
    if let _ = ad as? ANInterstitialAd {
      interstitialAdView?.display(from: self)
    }
  }
  
  func ad(_ ad: ANAdProtocol!, requestFailedWithError error: Error!) {
    NSLog("ad:requestFailedWithError: \(ad.placementId)")
  }
  
  func adWasClicked(_ ad: ANAdProtocol!) {
    NSLog("adWasClicked: \(ad.placementId)")
  }
  
  func adWillClose(_ ad: ANAdProtocol!) {
    NSLog("adWillClose: \(ad.placementId)")
  }
  
  func adDidClose(_ ad: ANAdProtocol!) {
    NSLog("adDidClose: \(ad.placementId)")
  }
  
  func adWillPresent(_ ad: ANAdProtocol!) {
    NSLog("adWillPresent: \(ad.placementId)")
  }
  
  func adDidPresent(_ ad: ANAdProtocol!) {
    NSLog("adDidPresent: \(ad.placementId)")
  }
  
  func adWillLeaveApplication(_ ad: ANAdProtocol!) {
    NSLog("adWillLeaveApplication: \(ad.placementId)")
  }
  
  //MARK: ANInterstitialAdDelegate
  
  func adFailed(toDisplay ad: ANInterstitialAd!) {
    NSLog("adFailedToDisplay: \(ad.placementId)")
  }
  
}
